### PR TITLE
improve: Update ethereum package versions

### PR DESCRIPTION
**Pull request by nicholaspai**
Two big features I'd like to take advantage of:
- improved `hardhat` base package performance
- `hardhat-etherscan > 3.0.0` allows you to specify multiple API keys
- I bumped the `typechain` type which produced a large diff, let me know if you want me to remove this part

### CODE REVIEW

**Review by mrice32 — COMMENTED** [review]
Looks good! One comment on typechain

File: package.json [hunk 1]
```diff
@@ -27,11 +27,11 @@
...
   "devDependencies": {
-    "@nomiclabs/hardhat-ethers": "^2.0.2",
-    "@nomiclabs/hardhat-etherscan": "^2.1.3",
-    "@nomiclabs/hardhat-waffle": "^2.0.0",
+    "@nomiclabs/hardhat-ethers": "^2.0.5",
+    "@nomiclabs/hardhat-etherscan": "^3.0.3",
+    "@nomiclabs/hardhat-waffle": "^2.0.3",
     "@typechain/ethers-v5": "^7.0.1",
-    "@typechain/hardhat": "^2.3.0",
+    "@typechain/hardhat": "^4.0.0",
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply package.json:34]: Yeah I'll hold off for now

**Review by mrice32 — CHANGES_REQUESTED** [review]
@nicholaspai I think you need to rerun yarn to regenerate yarn.lock from your last commit.

**Comment by nicholaspai** [comment]
> @nicholaspai I think you need to rerun yarn to regenerate yarn.lock from your last commit.

Great catch thanks

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**